pkg/server/metrics/metrics.go

```diff
@@ -87,41 +96,52 @@ func newServerMetrics() *ServerMetrics {
 	)
 	httpConnections := prometheus.NewGauge(
 		prometheus.GaugeOpts{
-			Namespace: namespace,
-			Subsystem: subsystem,
-			Name:      "http_connections",
+			Namespace: Namespace,
+			Subsystem: Subsystem,
+			Name:      HTTPConnectionsMetric,
 			Help:      "Number of current HTTP CONNECT connections",
 		},
 	)
 	backend := prometheus.NewGaugeVec(
 		prometheus.GaugeOpts{
-			Namespace: namespace,
-			Subsystem: subsystem,
-			Name:      "ready_backend_connections",
+			Namespace: Namespace,
+			Subsystem: Subsystem,
+			Name:      BackendConnectionsMetric,
 			Help:      "Number of konnectivity agent connected to the proxy server",
 		},
 		[]string{},
 	)
 	pendingDials := prometheus.NewGaugeVec(
 		prometheus.GaugeOpts{
-			Namespace: namespace,
-			Subsystem: subsystem,
-			Name:      "pending_backend_dials",
+			Namespace: Namespace,
+			Subsystem: Subsystem,
+			Name:      PendingDialsMetric,
 			Help:      "Current number of pending backend dial requests",
 		},
 		[]string{},
 	)
 	fullRecvChannels := prometheus.NewGaugeVec(
 		prometheus.GaugeOpts{
-			Namespace: namespace,
-			Subsystem: subsystem,
-			Name:      "full_receive_channels",
+			Namespace: Namespace,
+			Subsystem: Subsystem,
+			Name:      FullRecvChannelMetric,
 			Help:      "Number of current connections blocked by a full receive channel, partitioned by service method.",
 		},
 		[]string{
 			"service_method",
 		},
 	)
+	dialFailures := prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Namespace: Namespace,
+			Subsystem: Subsystem,
+			Name:      DialFailuresMetric,
+			Help:      "Number of dial failures observed. Multiple failures can occur for a single dial request.",
+		},
+		[]string{
+			"reason",
```

Comment:
Thoughts on adding `serverID` and `agentID` here?

Maybe `serverID` is redundant, because it always refers to the current proxy server publishing the metrics anyways. 

I guess with `agentID` we run the risk of increasing metrics cardinality because there's no limit on # of agents you could end up with, but generally it's not a huge number. But it would really help identify failure modes where there's a single bad agent.

Reply:
For a cloud provider the cardinality issue is huge. For a cluster the agentID and serverID may only be a handful. Across all clusters we expect no repeats, so this number gets very large, very quickly. In addition over time these IDs change (updates, restarts, ...) so again it grows without bound.

Reply:
Yeah, I agree with @cheftako. I think this is better tackled through logging.